[ext-authz] Add check logic for TCP principle (#29668)
istio/istio: samples/extauthz/README.md

as an example of integrating custom authorization system into Istio.

The Ext Authz server supports authorization check request using either HTTP (port 8000) or gRPC v2/v3 (port 9000) API and

> will allow the request if it includes the header `x-ext-authz: allow`. For TCP requests, the Ext Authz server will allow requests from source account `a`.

Reviewer: This is not specific to TCP, it also allows HTTP. And make it clear the the service account (not source account) `a` is just a default value for testing and can be configured explicitly with the flag `-allow_service_account`

> will allow the request if it includes the header `x-ext-authz: allow` or service account of the request is `a`. Here, `a` is just a default value for testing and can be configured explicitly with the flag -allow_service_account

Reviewer: ```
or if the service account of the source workload is `a`. Note `a` is just a default value for testing purpose and can be configured with the flag `-allow_service_account` when running the ext authz server.
```

as an example of integrating custom authorization system into Istio.

The Ext Authz server supports authorization check request using either HTTP (port 8000) or gRPC v2/v3 (port 9000) API and
will allow the request if it includes the header `x-ext-authz: allow` or if the service account of the source workload is `a`.

> Note `a` is just a default value for testing purpose and can be configured with the flag `-allow_service_account` when running the ext authz server.

Reviewer 2: ```suggestion
Note that `a` is just a default value for testing. It can be changed with the flag `-allow_service_account` when running the ext authz server.
```

Author: Thanks Frank for your review.
